mdatagen/go.mod

```diff
@@ -0,0 +1,49 @@
+module go.opentelemetry.io/build-tools/mdatagen
+
+go 1.19
+
+require (
+	github.com/google/go-cmp v0.5.9
+	github.com/stretchr/testify v1.8.2
+	go.opentelemetry.io/collector/component v0.78.2
+	go.opentelemetry.io/collector/confmap v0.78.2
+	go.opentelemetry.io/collector/pdata v1.0.0-rcv0012
+	go.opentelemetry.io/collector/receiver v0.78.2
+	go.opentelemetry.io/collector/semconv v0.78.2
```

Comment:
This is a good point, i hadn't considered moving it to the core repo as I considered this really just a build tool, but it may be a better location for this. WDYT @dmitryax

Reply:
> This looks like a "cross repo cyclic dependency" which may become a pain in maintenance. Have you conidered to keep this tool in Collector repo?

Unless I am missing something, since we only intend to use `mdatagen` as a CLI tool and not as a library, there would not be a cyclic dependency between two specific modules: the `internal/tools` module on the core collector repository would depend on `mdatagen` and `mdatagen` would depend in other, independent modules. I don't think this situation would introduce any issues; `internal/tools` dependencies are not really 'true dependencies' since we don't depend on any specific code and just do a blank import for dependency management.

Reply:
@mx-psi That is correct. It is not a "cyclic dependency between two specific modules". But if `mdatagen` would require changes in any of the `go.opentelemetry.io/collector/` modules then first changes would need to done on Collector repo (probably the collector should be released) and finally someone could make changes here. I find such model of work not efficient and I just kindly advise not going that way. This is what I call "cyclic repo dependency".

Reply:
I don't find that particularly concerning, but I guess it's more problematic until we have v1.0 on the modules used by mdatagen so I am fine with moving it to core and arguing again for this once we reach 1.0 :)